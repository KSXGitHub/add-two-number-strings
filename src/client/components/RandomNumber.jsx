import React from 'react'

export default class RandomNumber extends React.Component {
  render () {
    return <div>
      <span className='label'>Random Number: </span>
      <span className='value'>{Math.random()}</span>
    </div>
  }
}
