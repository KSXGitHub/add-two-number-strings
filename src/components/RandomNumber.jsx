import React from 'react'
import Paper from 'material-ui/Paper'

export default class RandomNumber extends React.Component {
  constructor (props) {
    super(props)
    const {display = String, init = Math.random, style = {}} = props
    const value = init()
    const displayValue = () => display(value)
    this.state = {displayValue, style}
  }

  render () {
    const {
      label: labelStyle = {},
      value: valueStyle = {}
    } = this.state.style

    return <Paper zDepth={0}>
      <span className='label' style={labelStyle}>
        Random Number:&nbsp;
      </span>
      <span className='value' style={valueStyle}>
        {this.state.displayValue()}
      </span>
    </Paper>
  }
}
