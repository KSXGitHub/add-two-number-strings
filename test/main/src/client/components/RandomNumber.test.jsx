import './__lib__/material-ui-mocks.jsx'
import React from 'react'
import renderer from 'react-test-renderer'
import RandomNumber from '../../../../../src/client/components/RandomNumber.jsx'
import DefaultMuiTheme from '../../../../../src/client/components/DefaultMuiTheme.jsx'

function WrappedRandomNumber (props) {
  return <DefaultMuiTheme>
    <RandomNumber {...props} />
  </DefaultMuiTheme>
}

describe('Snapshot: components/RandomNumber.jsx', () => {
  const init = () => 0.123456789

  test('with default display function', () => {
    const tree = renderer.create(<WrappedRandomNumber init={init} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with specified display function', () => {
    ; [
      value => JSON.stringify({value}),
      value => Number(value).toString(16).slice(2, 16),
      value => Number(value).toString(36).slice(2, 13),
      value => value * value,
      value => [0, 1, 2, 3].map(x => value * x).join(', ')
    ]
      .map(display => (<WrappedRandomNumber init={init} display={display} />))
      .map(element => renderer.create(element))
      .map(instance => instance.toJSON())
      .forEach(tree => expect(tree).toMatchSnapshot())
  })
})
