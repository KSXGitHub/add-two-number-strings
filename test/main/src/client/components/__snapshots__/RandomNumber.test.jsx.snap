// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/RandomNumber.jsx with default display function 1`] = `
<div>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    0.123456789
  </span>
</div>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 1`] = `
<div>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    {"value":0.123456789}
  </span>
</div>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 2`] = `
<div>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    1f9add3739635f
  </span>
</div>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 3`] = `
<div>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    4fzzzxjylrx
  </span>
</div>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 4`] = `
<div>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    0.01524157875019052
  </span>
</div>
`;

exports[`Snapshot: components/RandomNumber.jsx with specified display function 5`] = `
<div>
  <span
    className="label"
  >
    Random Number: 
  </span>
  <span
    className="value"
  >
    0, 0.123456789, 0.246913578, 0.370370367
  </span>
</div>
`;
